org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/scopes/BuildScopeUnit.scala

```diff
@@ -77,15 +77,16 @@ class BuildScopeUnit(val scope: CompileScope, val owningProject: IScalaProject,
     }
   }
 
-  private def areDependedUnitsBuilt = {
-    val wrongScopes = dependentUnitInstances filter { _.hasErrors } map { _.scope }
-    if (wrongScopes.nonEmpty) {
-      BuildProblemMarker.create(owningProject.underlying,
-        s"${owningProject.underlying.getName}'s ${scope.name} not built due to errors in dependent scope(s) ${wrongScopes.map(_.name).toSet.mkString(", ")}")
-      false
-    } else true
+  def isAnySourceDirAssignToScope: Boolean = sources.nonEmpty
```

Comment:
But it seems there is no good name... that is shorter than the implementation :) Seriously, this doesn't carry its own weight.